src/BSON/Javascript.c

```diff
@@ -215,6 +216,41 @@ PHP_METHOD(Javascript, getScope)
 }
 /* }}} */
 
+/* {{{ proto array Javascript::jsonSerialize()
+*/
+PHP_METHOD(Javascript, jsonSerialize)
+{
+	php_phongo_javascript_t *intern;
+	bool has_scope;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_JAVASCRIPT_OBJ_P(getThis());
+
+	has_scope = (intern->scope && intern->scope->len);
+	array_init_size(return_value, has_scope ? 2 : 1);
```

Comment:
I wouldn't bother with the trinary here. Just make it 2 - it's merely a hint to the underlying algorithm anyway. And in that case, you can get rid of the ```has_scope``` variable, and just do the test in the ```if()``` below.